cpp/src/slice2java/GenCompat.cpp

```diff
@@ -210,6 +210,28 @@ writeParamList(Output& out, vector<string> params, bool end = true, bool newLine
     }
 }
 
+// Returns the "length" of the constructor parameter list from the given data member list.
+// All types are counted as 1, except for long and double which are counted as 2.
+// See https://docs.oracle.com/javase/specs/jvms/se20/html/jvms-4.html#jvms-4.3.3
+int
+constructorParameterLength(const DataMemberList& members)
```

Comment:
We really don't have any shared code between slice2java and slice2java --compat??

Reply:
No

Reply:
You should move it to Slice/JavaUtil.h

Reply:
Right, I forgot about those. Will fix.